Lib/string.py

```diff
@@ -89,6 +66,25 @@ class Template(metaclass=_TemplateMetaclass):
     braceidpattern = None
     flags = _re.IGNORECASE
 
+    def __init_subclass__(cls):
+        super().__init_subclass__()
+        if 'pattern' in cls.__dict__:
+            pattern = cls.pattern
+        else:
+            pattern = r"""
```

Comment:
Perhaps this pattern ought to be put at module scope, just above the class definition?  The verbose representation might be more readable without all that leading whitespace.